src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.ReadObjectResponse.liquid

```diff
@@ -2,37 +2,83 @@
 
 protected virtual async System.Threading.Tasks.Task<ObjectResponseResult<T>> ReadObjectResponseAsync<T>(System.Net.Http.HttpResponseMessage response, System.Collections.Generic.IReadOnlyDictionary<string, System.Collections.Generic.IEnumerable<string>> headers)
 {
+    if (response.Content == null)
+    {
+        return new ObjectResponseResult<T>(default(T), string.Empty);
+    }
+
     if (ReadResponseAsString)
     {
         var responseText = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
-        try
+
+        if (response.Content.Headers.ContentType != null && response.Content.Headers.ContentType.MediaType.Contains("xml"))
         {
-            var typedBody = Newtonsoft.Json.JsonConvert.DeserializeObject<T>(responseText, JsonSerializerSettings);
-            return new ObjectResponseResult<T>(typedBody, responseText);
+            try
+            {
+                using (var reader = new System.IO.StringReader(responseText))
+                {
+                    var serializer = new System.Xml.Serialization.XmlSerializer(typeof(T));
+                    var typedBody = (T)serializer.Deserialize(reader);
+                    return new ObjectResponseResult<T>(typedBody, string.Empty);
+                }
+            }
+            catch (Exception exception) 
+            {
+                var message = "Could not deserialize the response body stream as " + typeof(T).FullName + ".";
+                throw new {{ ExceptionClass }}(message, (int)response.StatusCode, null, headers, exception);
+            }
         }
-        catch (Newtonsoft.Json.JsonException exception)
+        else
```

Comment:
If the code will check the content-type for XML responses, shouldn't it do the same for JSON responses?

Is there a way for a Swagger file to assert the response will be a JSON serialized object even if it's returned with an incorrect MIME type? (i.e. JSON served as `text/plain` or `text/html`?)